### ISSUE TITLE
Investigate options for interfacing with the electronic knitting machine

### ISSUE THREAD

**Issue by JackiePease** [DoESLiverpool/somebody-should#1341]
We have a new (to us) electronic knitting machine which has been donated by Sam Meech. It's a Brother Electroknit KH-900. 
There are various interface options we could use in order to send patterns from computers etc.
The AYAB interface is well known: https://ayab-knitting.com/ayab-interface/ https://github.com/AllYarnsAreBeautiful/ayab-hardware/tree/master/Shieldv2x
This is open source hardware so we could buy one or build one ourselves

**Comment by jackie1050** [comment]
I ordered a shield from Etsy: www.etsy.com/uk/listing/574430686/kh-900965-ayab-shield-kit-v14-th-with?ref=shop_home_active_1
Estimated delivery 3 -14 Jan
Donated to DoES Liverpool

**Comment by jackie1050** [comment]
The shield has arrived - I've left it behind the knitting machine for the time being. There are lots of components that will need to be soldered.

**Comment by jackie1050** [comment]
@paulgeering has soldered all the components to the board. 
We have used the Arduino Uno from the Knitic, so it is important to download new software before plugging it into the AYAB board.
We tried to download the AYAB software for the laptop from https://manual.ayab-knitting.com/installation/software/ but ran into the usual Python version problems, so will try again on Saturday.

**Comment by jackie1050** [comment]
I continued to look into this at the Ladies of Code meetup with Rae and Jax Steele, and again today.
Rae managed to produce a 2 colour pattern using the original card, so we know that the machine is working.
There are some reasonable casting on instructions here: https://m.youtube.com/watch?v=L6D5O0mhaAU
We had a lot of issues with loops of yarn being formed at the ends of rows and getting tangled in the K carriage (if this happens, then turn the dial on the carriage to the CR position, which will let you lift it off, as otherwise there is a risk of damage to the needles). I've tweeted Sam and Victoria to see if they have any suggestions, and @RussCoty is going to check with his mum.
We didn't get much chance to test the AYAB interface, but it does connect and the buzzer works. It looks like for the KH900, it can be powered through the Arduino and doesn't need a separate power cable.
There seems to be an interesting tutorial here:
https://xxxclairewilliamsxxx.wordpress.com/hack-ta-machine-a-tricoter/hack-your-knitting-machine-tutorial/
